File: test_cart.js
const fs = require('fs');

async function test() {
  const loginRes = await fetch("http://localhost:8888/auth/standard-login", {
    method: "POST",
    headers: {"Content-Type": "application/x-www-form-urlencoded"},
    body: "email=admin%40poool.finance&password=TestPass123%21",
    redirect: "manual"
  });
  
  const cookies = loginRes.headers.get("set-cookie");
  
  const cartRes = await fetch("http://localhost:8888/cart", {
    headers: { "cookie": cookies }
  });
  const html = await cartRes.text();
  fs.writeFileSync("/Users/martin/Projects/poool/cart_out.html", html);
  console.log("Written to cart_out.html. Summary box present?", html.includes('id="cart-page-summary"'));
}
test();

File: test_cart.py
import urllib.request
import urllib.parse
from http.cookiejar import CookieJar

cj = CookieJar()
opener = urllib.request.build_opener(urllib.request.HTTPCookieProcessor(cj))

# Login
login_data = urllib.parse.urlencode({'email': 'admin@poool.finance', 'password': 'TestPass123!'}).encode('utf-8')
try:
    opener.open('http://localhost:8888/auth/standard-login', data=login_data)
except Exception as e:
    print("Login err:", e)

# Fetch cart
cart_req = urllib.request.Request('http://localhost:8888/cart')
try:
    resp = opener.open(cart_req)
    html = resp.read().decode('utf-8')
    with open('/Users/martin/Projects/poool/cart_out.html', 'w') as f:
        f.write(html)
    print("Saved to cart_out.html")
except Exception as e:
    print("Cart err:", e)
